src/main/java/com/microsoft/graph/Requests/BaseClient.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.graph.Requests;
+
+import com.microsoft.graph.content.BatchRequestBuilder;
+import com.microsoft.kiota.RequestAdapter;
+import com.microsoft.kiota.authentication.AuthenticationProvider;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.OkHttpClient;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Default client implementation.
+ */
+public class BaseClient implements IBaseClient{
```

Comment:
I don't believe we can have a base client (class) in core anymore.
That's because the GeneratedGraphServiceClient will be a class that doesn't inherit from this one and Java doesn't support multiple inheritance.
So this will either have to move to the service library and inherit from the generated client or be removed all together.

Reply:
Hey Vincent, it seems that in C# there is this baseClient class as a means to create a lightweight client that doesn't rely on the service library. I believe we can keep this class here as GraphServiceClient will extend from BaseGraphServiceClient and implement IBaseClient. This is the same pattern that we have in C#. This means I will have to create the GraphServiceClient following the same pattern that is already in C#, but I believe we can keep both.

Reply:
alright, but just to be clear, this one won't be in the inheritance structure of a client for somebody using the graph service client from the service lib, correct?

Reply:
no, the the GraphServiceClient will extend the BaseGraphServiceClient and implement IBaseClient, GraphServiceClient will not use any BaseClient implementations.

Reply:
so should this class be removed from the PR?